src/NanoVG/Internal/FFIHelpers.hs

```diff
@@ -23,15 +25,25 @@ withCString t = useAsCString (T.encodeUtf8 t)
 
 -- | Wrapper around 'useAsCStringLen' that uses 'CUChar's
 useAsCStringLen' :: ByteString -> ((Ptr CUChar,CInt) -> IO a) -> IO a
-useAsCStringLen' bs f = useAsCStringLen bs (\(ptr,len) -> f (castPtr ptr,fromIntegral len))
+useAsCStringLen' bs f = useAsCStringLen bs ((\(ptr,len) -> return (castPtr ptr,fromIntegral len)) >=> copyCStringLen >=> f)
+  where
+    -- | Copy memory under given pointer no a new address.
+    -- The allocated memory is not garbage-collected and needs to be freed manually later.
+    copyCStringLen :: Integral b => (Ptr a, b) -> IO (Ptr a, b)
+    copyCStringLen (from, len) =
+      let intLen = fromIntegral len
+      in do
+        to <- mallocBytes intLen
```

Comment:
Can you please add a comment that this is freed by NanovVG as part of `nvgDeleteGL3`?

Reply:
Done. Feel free to change it to your liking if you think it could be improved.

Also thanks for maintenance and looking at it rather promptly!